src/Sulu/Bundle/AdminBundle/Resources/js/components/AutoCompletePopover/Suggestion.js

```diff
@@ -24,21 +24,23 @@ export default class Suggestion extends React.PureComponent<Props> {
             return null;
         }
 
-        const query = this.props.query || '';
-        const regex = new RegExp(query, 'gi');
-        const matches = text.match(regex);
-
-        if (!matches || query.length === 0) {
+        if (!this.props.query) {
             return text;
         }
 
-        let matchIndex = 0;
-        const highlightedMatches = text.replace(regex, () => {
-            return `<strong>${matches[matchIndex++]}</strong>`;
-        });
+        let matcher;
+        try {
+            // try to match all highlighted parts using case insensitive regular expression
+            matcher = new RegExp(this.props.query, 'gi');
+        } catch (e) {
+            // fallback to highlight first exact match if given query is an invalid regular expression like "*"
+            matcher = this.props.query;
+        }
+
+        const highlightedText = text.replace(matcher, '<strong>$&</strong>');
```

Comment:
unfortunately, we cannot easily use `replaceAll` here because it is not supported before `Node 15`:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll

Reply:
`replace` will highlight all matches if the query is a valid regular expression. if it is not a valid regular expression, only the first exact match will be highlighted. 

i think this is okay, because the second case happens very rarely (it is the same case that crashed the application until now).

Reply:
can we have different babel settings for tests? or import the core-js polyfill only in the tests? https://github.com/zloirock/core-js ?

Reply:
not sure about babel, but i think we could import the polyfill inside of the `testSetup.config.js` 🙂 
but as far as i see, `core-js` is not declared anywhere in our dependencies. i think i would need to add it?

Reply:
We could add it as test dependency if we did not yet need it anywhere. But I'm also then fine to keep it with this solution if all works and testa are green.

Reply:
maybe we could set another babel configuration using the jest `transform` option:
https://jestjs.io/docs/configuration#transform-objectstring-pathtotransformer--pathtotransformer-object

will have another looks at this when i find the time

Reply:
it looks like there is a bug in babel and babel does not transpile `String.replaceAll()` at all at the moment:
https://github.com/babel/babel/issues/13701

because of this, we cannot solve this by using a different babel config for tests 😕

Reply:
I think currently babel is not correctly configured as I think we should use `usage` in the usebuiltins babel configuration.
https://babeljs.io/docs/en/babel-preset-env#usebuiltins-usage

Reply:
thats right! i created a separate pull request for this: https://github.com/sulu/sulu/pull/6241

Reply:
thats right! i created a separate pull request for this: https://github.com/sulu/sulu/pull/6241

Reply:
i finally managed to change this to `replaceAll` 🥳